linetools/spectra/xspectrum1d.py

```diff
@@ -963,6 +963,97 @@ def stitch(self, idx=None, scale=1.):
         # Return
         return spec
 
+    def get_local_s2n(self, wv0, npix=51, flux_th=0., debug=False):
+        """It computes the local average signal-to-noise (s2n) over npix pixels around wv0.
+        If `flux_th` is given, pixels with fluxes below spec.flux*flux_th are masked out.
+
+        Parameters
+        ----------
+        wv0 : Quantity
+            Observed wavelength where to perform the calculation.
+        npix : float or int, optional
+            Number of pixels to perform the calculation (does not
```

Comment:
Always assess 51 pixels (when possible), so the range is increased if there are masked pixels.